File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.py
class Solution:
  def maxProfit(self, prices: List[int]) -> int:
    ans, l, r = 0, 0, 1
    while r < len(prices):
      if prices[r] >= prices[l]:
        while r < len(prices)-1 and prices[r+1] >= prices[r]:
          r += 1
        ans += prices[r]-prices[l]
      l = r
      r += 1
    return ans

File: 0122-best-time-to-buy-and-sell-stock-ii/0122-best-time-to-buy-and-sell-stock-ii.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  let [ans, l, r] = [0, 0, 1];
  while (r < prices.length) {
    if (prices[r] >= prices[l]) {
      while (r < prices.length - 1 && prices[r + 1] >= prices[r]) r++;
      ans += prices[r] - prices[l];
    }
    l = r;
    r++;
  }
  return ans;
};